### ISSUE TITLE
Date32 datatype is not supported in dictionaries

### ISSUE THREAD

**Issue by vpanfilov** [ClickHouse/ClickHouse#32913]
Date32 datatype is not supported in dictionaries. For example:
```sql
CREATE DICTIONARY testDate32
(
  date Date32
)
PRIMARY KEY date
SOURCE(CLICKHOUSE(
  host 'localhost'
  query 'SELECT toDate32(now())'
))
LAYOUT(COMPLEX_KEY_HASHED())
LIFETIME(0)
```

```
DB::Exception: Unknown type Date32 for dictionary attribute. (UNKNOWN_TYPE) (version 21.12.2.17 (official build))
```

**Comment by kitaisreal** [comment]
Cannot be implemented until #23194.

**Comment by alexey-milovidov** [comment]
@kitaisreal Need to resubmit the PR.
